src/devices/Pn532/Pn532.cs

```diff
@@ -123,7 +124,8 @@ public Pn532(string portName)
             bool ret = SetSecurityAccessModule();
             _logger.LogInformation($"Setting SAM changed: {ret}");
             // Check the version
-            if (!IsPn532())
+            var ver = IsPn532();
+            if (checkVersion && !ver)
```

Comment:
```suggestion
            if (checkVersion && !IsPn532())
```
No need to have the local variable and no need to check the value if checkVersion is false. If you simplify the ifcheck you fix both issues.

Reply:
According to the description in the PR (see above), the request for the version is still required, only the result shall be ignored.

Reply:
I confirm from the description and from what I found, you have to call the the check version even if the result should be ignored. Otherwise, seems like the chips doesn't then work properly. It's like initialization, it seems to be required.

Reply:
Oh sorry I missed that part of the description. In that case, my suggestion would be to use my suggestion except reverse the order of the operands so that the check always happens but you won't need the extra variable. So like:

```c#
// Checking version first in order to initialize the device
If (!IsPN532() && checkVersion)
```

Reply:
Yes, fair! Thanks for the suggestion. Adjusted.